src/main/scala/org/broadinstitute/hail/driver/AnnotateVariantsLoci.scala

```diff
@@ -0,0 +1,77 @@
+package org.broadinstitute.hail.driver
+
+import org.apache.spark.sql.Row
+import org.broadinstitute.hail.Utils._
+import org.broadinstitute.hail.annotations._
+import org.broadinstitute.hail.expr._
+import org.broadinstitute.hail.utils._
+import org.broadinstitute.hail.variant.{Locus, Variant}
+import org.kohsuke.args4j.{Argument, Option => Args4jOption}
+
+import scala.collection.JavaConverters._
+
+object AnnotateVariantsLoci extends Command with JoinAnnotator {
+
+  class Options extends BaseOptions with TextTableOptions {
+    @Argument(usage = "<files...>")
+    var arguments: java.util.ArrayList[String] = new java.util.ArrayList[String]()
+
+    @Args4jOption(required = false, name = "-r", aliases = Array("--root"),
+      usage = "Period-delimited path starting with `va' (this argument or --code required)")
+    var root: String = _
+
+    @Args4jOption(required = false, name = "-c", aliases = Array("--code"),
+      usage = "Use annotation expressions to join with the table (this argument or --root required)")
+    var code: String = _
+
+    @Args4jOption(required = true, name = "-e", aliases = Array("--locus-expr"),
+      usage = "Specify an expression to construct a locus")
+    var locusExpr: String = _
+  }
+
+  def newOptions = new Options
+
+  def name = "annotatevariants loci"
+
+  def description = "Annotate variants by locus (chromosome, position) with delimited text file"
+
+  def supportsMultiallelic = true
+
+  def requiresVDS = true
+
+  def run(state: State, options: Options): State = {
+
+    val files = hadoopGlobAll(options.arguments.asScala, state.hadoopConf)
+    if (files.isEmpty)
+      fatal("Arguments referred to no files")
+
+    val vds = state.vds
+
+    val (expr, code) = (Option(options.code), Option(options.root)) match {
+      case (Some(c), None) => (true, c)
+      case (None, Some(r)) => (false, r)
+      case _ => fatal("this module requires one of `--root' or `--code', but not both")
+    }
```

Comment:
I think we should keep things consistent for now.  I can make another PR to add deprecation warnings for all --root args, but I'd like this module to have similar usage to `annotatevariants table`